### PR TITLE
Plan Editor: Option to upload plans from a spreadsheet

### PR DESCRIPTION
**Pull request by dmgav**
Extention of a Plan Editor Widget: implementation of an option that allows uploading plans from a spreadsheet. Plan Editor widget is now contains an additional button 'Batch Upload':
image

Clicking on the `Batch Upload` button opens `Batch Upload` dialog box:
image

The dialog box contains the button `..`, which opens standard dialog for selection of an existing file and optional combo box for selecting spreadsheet type. The list of spreadsheet types (strings) may be set during initialization of the model. If the list is not set, then the combo box is not displayed:
image

Once the file is selected, the `Ok` button is enabled.
image

Pressing the `Ok` button adds the plans to the queue. The block of newly inserted items is automatically selected so that queue operations (copy/delete/move) could be applied to the block.
image

### CODE REVIEW

File: bluesky_widgets/models/run_engine_client.py [hunk 1]
```diff
@@ -58,6 +61,13 @@ def __init__(self, zmq_server_address=None, user_name="GUI Client", user_group="
         #   using positions is consistent. Empty list - no items are selected
         self._selected_history_item_pos = []
 
+        # Current directory (used for opening/saving files).
+        self.current_dir = None
```
Comment (danielballan) [inline bluesky_widgets/models/run_engine_client.py:65]: Should this set to `os.cwd()` by default?

File: bluesky_widgets/models/run_engine_client.py [hunk 2]
```diff
@@ -58,6 +61,13 @@ def __init__(self, zmq_server_address=None, user_name="GUI Client", user_group="
...
+        # Set the variable to the name of the Queue Server Custom Module (if available)
+        self.qserver_custom_module_name = None
```
Comment (danielballan) [inline bluesky_widgets/models/run_engine_client.py:67]: Should this be a list of modules?

Comment (dmgav) [reply bluesky_widgets/models/run_engine_client.py:67]: No, just the name of the module or None if there is no such module. I am using environment variable to pass the module name to HTTP server.

File: bluesky_widgets/models/run_engine_client.py (hunk 1 above)
Comment (dmgav) [reply bluesky_widgets/models/run_engine_client.py:65]: It is set to `os.cwd()` later. I think general flow of data and setting should be revisited at certain point to take into account the most of the possible use cases.

**Pull request merged**